ts/src/pro/bitfinex2.ts

```diff
@@ -277,13 +277,13 @@ export default class bitfinex2 extends bitfinex2Rest {
             const limit = this.safeInteger (this.options, 'tradesLimit', 1000);
             this.myTrades = new ArrayCacheBySymbolById (limit);
         }
-        const array = this.myTrades;
-        array.append (trade);
-        this.myTrades = array;
+        const tradesArray = this.myTrades;
```

Comment:
@ttodua for instance now this is transpiled to 
```
        $array = $this->myTrades;
        $array->append ($trade);
        $this->myTrades = $array;
```
is there an issue with it?

Reply:
but it's vulnerable code, independent from current result. it might misbehave if the body of that method is changed. so, we should have a thumb rule to avoid such reserved words for arrays (at least, even in best case results, it changed variable names will not hurt...)

Reply:
@ttodua sorry I'm still not 100% clear yet, can you give a concrete example?

Reply:
> @ttodua sorry I'm still not 100% clear yet, can you give a concrete example?

for example this - https://discord.com/channels/690203284119617602/1073151528753434624/1146114079488999484

Reply:
@ttodua but is this issue related to the AST-transpiler then?

Reply:
> @ttodua but is this issue related to the AST-transpiler then?

no, it's not related to AST-transpiler, because those glitches are currently happening because of ccxt's custom transpiler imperfection.

Reply:
so, one line above of them, if you place `const xyz = Object.keys (a);` and re-transpile, you will see such in php:
```
$keys = is_array($query) ? array_keys($query) : $array();
```
so, we shouldn't name variable reserved words, to avoid possible complications.

Reply:
sorry, I don't understand the last comment. what I mean is that, we shouldnt have `const array = ` anywhere across lib, because where it is , that method becomes fragile, and we better to avoid this and there is no hurt in that, other than leaving it and having fragile code. 
the thing is that, `Object.keys ()` will be transiled wrongly if in method body there is `const array = ` so, we better to avoid that fragile definition across lib.

Reply:
but sure, as you wish, just lmk and we can close this PR

Reply:
![image](https://github.com/ccxt/ccxt/assets/43336371/47594714-a604-4ac6-adf3-cca338719b57)

@ttodua actually it is more than that, sometimes we can even break the PHP code